CRM/Contact/BAO/Group.php

```diff
@@ -1378,4 +1378,27 @@ public static function getChildGroupIds($regularGroupIDs) {
     return $childGroupIDs;
   }
 
+  /**
+   * Check parent groups and filter out the disabled ones.
+   *
+   * @param array $parentArray
+   *   Array of group Ids.
+   *
+   * @return int
+   */
+  public static function filterActiveGroups($parentArray) {
+    if (count($parentArray) > 1) {
+      foreach ($parentArray as $key => $groupId) {
+        $isActive = civicrm_api3('Group', 'getvalue', array(
+          'id' => $groupId,
+          'return' => 'is_active',
+        ));
+        if (!$isActive) {
+          unset($parentArray[$key]);
+        }
+      }
```

Comment:
How big is `$parentArray` likely to be? We could be fetching this data in a single api call rather than looping through each and hitting the database multiple times.